modules/entity_version_history/src/Form/HistoryTabSettingsForm.php

```diff
@@ -0,0 +1,209 @@
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'entity_version_history_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   * @SuppressWarnings(PHPMD.NPathComplexity)
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $bundle_labels = [];
+    $entity_labels = [];
+    $field_labels = [];
+    $history_configs = [];
+    // Get entity types and bundles where the entity_version field is present.
+    $versioned_entity_types = $this->entityFieldManager->getFieldMapByFieldType('entity_version');
+
+    foreach ($versioned_entity_types as $entity_type_id => $fields) {
+      $definition = $this->entityTypeManager->getDefinition($entity_type_id);
+      if (!$definition instanceof ContentEntityTypeInterface) {
+        // We are only interested in the content entity types.
+        continue;
+      }
+      if (!$definition->hasLinkTemplate('canonical') || !$definition->hasHandlerClass('moderation')) {
+        // We are only interested in the content entity types has canonical
+        // links and are moderated.
+        continue;
+      }
+
+      // We need a list of options with labels for the form checkboxes.
+      $entity_labels[$entity_type_id] = $definition->getLabel() ?: $entity_type_id;
+
+      foreach ($fields as $field_name => $bundle_info) {
+        foreach ($bundle_info['bundles'] as $bundle_name) {
+          // We need load up the bundle and the field to prepare the checkboxes
+          // with values and labels.
+          $bundle = $this->entityTypeManager->getStorage($definition->getBundleEntityType())->load($bundle_name);
+          $bundle_labels[$entity_type_id][$bundle_name] = $bundle->label();
+          $field_config = $this->entityTypeManager->getStorage('field_config')->load("$entity_type_id.$bundle_name.$field_name");
+          $field_labels[$entity_type_id][$bundle_name][$field_name] = $field_config->label();
+
+          if ($config = HistoryTabSettings::loadByEntityTypeBundle($entity_type_id, $bundle_name)) {
+            // Get the existing configs to pre-fill the form fields with
+            // default values.
+            $history_configs[$entity_type_id][$bundle_name][$field_name] = $config->getTargetField();
+          }
+        }
+      }
+    }
+
+    asort($entity_labels);
+
+    // Create checkboxes for all entity types.
+    $form['entity_types'] = [
+      '#title' => $this->t('History tab settings'),
+      '#type' => 'checkboxes',
+      '#options' => $entity_labels,
+      '#default_value' => is_array($history_configs) ? array_keys($history_configs) : [],
+    ];
+
+    // Create checkboxes for all bundles.
+    foreach ($bundle_labels as $entity_type_id => $bundles) {
+      $form['settings'][$entity_type_id] = [
```

Comment:
shall we wrap the entity type - bundles elements into a details or fieldset to make the form a bit more readable? With multiple entity types and bundles it becomes a mess.